### PR TITLE
Support method naming convention

### PR DESCRIPTION
**Pull request by webhacking**
Add a method naming convention rule.
e.g. send_notification_post, sendNotificationPost

### CODE REVIEW

File: application/config/rest.php [hunk 1]
```diff
@@ -604,3 +604,15 @@
...
 $config['allowed_cors_origins'] = [];
+
+
+/*
+|--------------------------------------------------------------------------
+| Choose method naming convention
+|--------------------------------------------------------------------------
+|
+| Default value is `snake_case`
+| e.g. send_notification_post, sendNotificationPost
```
Comment (Hanisch-IT) [inline application/config/rest.php]: You should add the posible configuration options 
- camelcase
- snake_case

And `snake_case` should be `snakecase` oder `camelcase` should be `camel_case`

So the style of the optins are the same

File: application/libraries/REST_Controller.php [hunk 2]
```diff
@@ -653,6 +653,32 @@ protected function preflight_checks()
...
+    * @method define_controller_method
+    * @param  string $object_called
+    * @return string rename controller method
+    */
+	private function define_controller_method($object_called)
+	{
+		$rename_method = $object_called.'_'.$this->request->method;
+		$method_naming_convention = $this->config->item('method_naming_convention');
+
+		switch ( strtolower(str_replace(['_','-'], '', $method_naming_convention)) ) {
```
Comment (Hanisch-IT) [inline application/libraries/REST_Controller.php]: `strtolower(str_replace(['_','-'], '', $method_naming_convention)`should not be needed. You cant avoid all the possibilities what the user can configure wrong.
Write the possibilities only down in the `rest.php`.

Else you have to care about all the wrong configurations and check if you can auto correct it. That overloads the library. [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)

**Comment by webhacking** [comment]
Thank you for your feedback.
This library is really useful. I just wanted to raise issue regarding naming convention. So I thought it would be better to offer directions about naming convention to users.

However, I do not understand your comment. I thought `snake_case` should have underline and `camelcase` should not. I have attached the link, so you can take a look at it. 

link: https://en.wikipedia.org/wiki/Snake_case

**Comment by Hanisch-IT** [comment]
You are right. 

`snake_case` is totaly okay. Thank your for your changes and your effort.

**Comment by chriskacerguis** [comment]
Thanks for submitting this.  As this library is geared towards Code Igniter and per their style guide; the standard is snake_case, could you please help me to understand what benefit this would bring?  😄

**Comment by chriskacerguis** [comment]
Haven't gotten a response.  Closing.

**Pull request closed without merging**